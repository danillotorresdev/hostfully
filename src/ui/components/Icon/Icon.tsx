const url = "/assets/sprite.svg";

type SvgNameMapper =
  | "arrow-right"
  | "arrow-left"
  | "trash"
  | "edit"
  | "logo"
  | "calendar"
  | "departure"
  | "arrival"
  | "arrow-foward"
  | "arrow-back"
  | "members"
  | "member"
  | "close";

type IconProps = {
  name: SvgNameMapper;
  size?: "sm" | "md" | "lg";
  color?: string;
};

const SVG_SIZE_MAPPER = {
  sm: 16,
  md: 24,
  lg: 32,
};

export const Icon = ({ name, size = "md", color }: IconProps) => {
  const svgSize = SVG_SIZE_MAPPER[size];

  return (
    <svg
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
      className={`svg svg-${name}`}
      data-testid="icon-svg"
      style={{
        width: svgSize,
        height: svgSize,
        color,
      }}
    >
      <use xlinkHref={`${url}#${name}`} />
    </svg>
  );
};
