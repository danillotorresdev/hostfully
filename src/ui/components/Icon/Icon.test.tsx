import { Icon } from "./Icon";
import { describe, expect, it } from "vitest";
import { render, screen } from "@/utils/settings/tests/utilities";

describe("Icon component", () => {
  it("renders icon with correct attributes", () => {
    render(<Icon name="arrow-right" size="md" color="blue" />);

    const iconSvg = screen.getByTestId("icon-svg");

    expect(iconSvg).toMatchSnapshot();
  });
});
