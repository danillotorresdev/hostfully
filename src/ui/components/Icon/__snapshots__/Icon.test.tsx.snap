// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon component > renders icon with correct attributes 1`] = `
<svg
  class="svg svg-arrow-right"
  data-testid="icon-svg"
  style="width: 24px; height: 24px; color: blue;"
  viewBox="0 0 1024 1024"
  xmlns="http://www.w3.org/2000/svg"
>
  <use
    xlink:href="/assets/sprite.svg#arrow-right"
  />
</svg>
`;
